### PR TITLE
collada-dom 2.4.3

### PR DESCRIPTION
**Pull request by mikepurvis**
This change also switches the source from the Sourceforge-hosted releases which include binaries to just using the tagged source bundle from Github. It's a smaller download and gets away from Sourceforge's hosting issues.

@rdiankov, can you confirm that your repo is the upstream for collada-dom?

### CODE REVIEW

**Comment by mikepurvis** [comment]
Ack, sorry, bad committer name, will fix.

**Comment by dunn** [comment]
Any chance of a test as well? Thanks!

**Comment by mikepurvis** [comment]
Done.

File: Library/Formula/collada-dom.rb [hunk 1]
```diff
@@ -3,33 +3,45 @@
 class ColladaDom < Formula
   desc "C++ library for loading and saving COLLADA data"
   homepage 'http://www.collada.org/mediawiki/index.php/Portal:COLLADA_DOM'
```
Comment (vszakats) [inline Library/Formula/collada-dom.rb]: Suggesting to follow the redirect and move to 'https://':
https://www.khronos.org/collada/wiki/Portal:COLLADA_DOM

File: Library/Formula/collada-dom.rb [hunk 2]
```diff
@@ -3,33 +3,51 @@
...
-end
 
-__END__
-diff --git a/CMakeLists.txt b/CMakeLists.txt
-index 72b6deb..0c7f7ce 100644
---- a/CMakeLists.txt
-+++ b/CMakeLists.txt
-@@ -100,7 +100,7 @@ endif()
+  def test
+    # Test per: https://www.khronos.org/collada/wiki/DOM_guide:_Creating_documents#Creating_a_simple_document
```
Comment (dunn) [inline Library/Formula/collada-dom.rb]: We can probably get by without the comment (unless attribution is absolutely required!)

Comment (mikepurvis) [inline Library/Formula/collada-dom.rb]: Sure, I can get rid of it— just seemed reasonable to indicate to a future maintainer where the test came from (for example, if it breaks and they want to consult the same doc for the API change).

**Comment by mikepurvis** [comment]
Homepage URL updated, test comment removed.

File: Library/Formula/collada-dom.rb [hunk 3]
```diff
@@ -2,34 +2,50 @@
 
 class ColladaDom < Formula
   desc "C++ library for loading and saving COLLADA data"
-  homepage 'http://www.collada.org/mediawiki/index.php/Portal:COLLADA_DOM'
-  url 'https://downloads.sourceforge.net/project/collada-dom/Collada%20DOM/Collada%20DOM%202.4/collada-dom-2.4.0.tgz'
-  sha1 '74e28d670497abc897c06a41df7d28eea2bac836'
+  homepage 'https://www.khronos.org/collada/wiki/Portal:COLLADA_DOM'
+  url 'https://github.com/rdiankov/collada-dom/archive/v2.4.3.tar.gz'
+  sha1 '0eb3b9f3183de8e5183f509311aaa7e8bc2679b7'
```
Comment (DomT4) [inline Library/Formula/collada-dom.rb]: This should be `sha256`.

**Comment by tdsmith** [comment]
Can we avoid using cmake in the test? Otherwise cmake will have to become a runtime dependency, not a :build dependency.

**Comment by mikepurvis** [comment]
Is there a way to specify a test-only dependency? I'd prefer to test with CMake, as that verifies the find module, which is most of what I care about.

Alternatively, is there a good example of a pkgconfig-based test?

**Comment by tdsmith** [comment]
There's no concept of test dependencies. Using pkg-config isn't better since we'd have to make _that_ a dependency. Our goal for the `test` block is to have a minimal smoke test that makes sure the bottles are linked correctly. If we can just build this with `clang -I#{include} test.c -L#{lib} -lfoo -o test` that would be best.

**Comment by mikepurvis** [comment]
How would we manage the boost dependency in that scenario?

**Comment by tdsmith** [comment]
By hand; adding `Formula["boost"].opt_include` and `Formula["boost"].opt_lib` to the respective search paths should work.

**Comment by mikepurvis** [comment]
Hash updated, test updated to remove cmake dependency.

**Comment by mikepurvis** [comment]
Test block updated, getting a green light from brew-bot now. Did you want this squashed before merging?

**Comment by dunn** [comment]
Yes, please!

**Comment by mikepurvis** [comment]
Aight, rebase complete.

**Comment by dunn** [comment]
> @rdiankov, can you confirm that your repo is the upstream for collada-dom?

Has this been confirmed?

**Comment by mikepurvis** [comment]
Will ping by email. In any case, it's fairly clear that the old svn repo at SF is now dead, despite being referenced on this project's wiki pages:

https://www.khronos.org/collada/wiki/DOM_guide:_Setting_up#Downloading_the_COLLADA_DOM

**Comment by DomT4** [comment]
This needs rebasing against an up-to-date master branch. Any response from upstream yet on the upstream repo change?

**Comment by DomT4** [comment]
Closing here for now. If we can get upstream confirmation of the move we'll review a fresh PR on these changes. Thank you for the submission nonetheless :heart:

**Pull request closed without merging**